pkg/schemaconv/smd.go

```diff
@@ -122,6 +122,56 @@ func (c *convert) makeRef(model proto.Schema) schema.TypeRef {
 	return tr
 }
 
+func makeUnion(extensions map[string]interface{}) (*schema.Union, error) {
+	union := schema.Union{
+		Fields: map[string]string{},
+	}
+
+	if idiscriminator, ok := extensions["x-kubernetes-oneof-discriminator"]; ok {
+		discriminator, ok := idiscriminator.(string)
+		if !ok {
+			return nil, fmt.Errorf(`"x-kubernetes-oneof-discriminator" is of invalid type: %T`, idiscriminator)
+		}
+		union.Discriminator = &discriminator
+	}
+
+	if ifields, ok := extensions["x-kubernetes-oneof-fields"]; ok {
+		fields, ok := ifields.(map[interface{}]interface{})
+		if !ok {
+			return nil, fmt.Errorf(`"x-kubernetes-oneof-fields" is of invalid type: %T`, ifields)
+		}
+		for ifield, value := range fields {
+			field, ok := ifield.(string)
+			if !ok {
+				return nil, fmt.Errorf(`"x-kubernetes-oneof-fields": field is of invalid type: %T`, ifield)
+			}
+			discriminated, ok := value.(string)
+			if !ok {
+				return nil, fmt.Errorf(`"x-kubernetes-oneof-fields"/%v: value is of invalid type: %T`, field, value)
+			}
+			union.Fields[field] = discriminated
+		}
+	}
+
+	if union.Discriminator != nil && len(union.Fields) == 0 {
+		return nil, fmt.Errorf("discriminator set to %v, but no fields in union", *union.Discriminator)
```

Comment:
Is this actually an error?

Reply:
How does one indicate whether or not "None" is a permitted value?

Reply:
> Is this actually an error?

What's the point? if the only thing you have a discriminator, then it's just an enum. I would suspect that they happen because of a user-error.